modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -315,15 +321,31 @@ void finishHim(Exception failure, List<Failure> indexingFailures, List<SearchFai
 
             // id and source always come from the found doc. Scripts can change them but they operate on the index request.
             index.id(doc.getId());
-            index.source(doc.getSource());
+
+            // the source xcontent type and destination could be different
```

Comment:
Here is the issue I see. The ReindexRequest allows a source and destination request object to be set. The destination could have called setXContentType and had a script get applied. If the script gets applied then the source is set again as a map in `ScriptApplier`. This makes a conversion from the map to xcontent bytes based on the value of contentType in the index request. So there is an implicit conversion there. Now if you do not apply a script the data is set as it came back and even though a user had tried to tell us to change the format, we didn't. None of this is ideal, I just tried to make it better.